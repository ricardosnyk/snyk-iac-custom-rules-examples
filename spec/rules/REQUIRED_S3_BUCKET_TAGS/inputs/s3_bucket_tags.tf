provider "aws" {
  region = "us-east-1"
}

resource "aws_s3_bucket" "valid" {
  bucket = "valid-bucket"

  tags = {
    owner           = "devteam1"
    classification  = "public"
  }
}

resource "aws_s3_bucket" "invalid" {
  bucket = "invalid-bucket"

  tags = {
    owner           = "devteam5"
    classification  = "available"
  }
}