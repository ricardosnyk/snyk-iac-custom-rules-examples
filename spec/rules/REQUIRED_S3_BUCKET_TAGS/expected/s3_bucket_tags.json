[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "aws_s3_bucket.invalid",
    "resource_namespace": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
    "resource_type": "aws_s3_bucket",
    "severity": "high",
    "resources": [
      {
        "id": "aws_s3_bucket.invalid",
        "type": "aws_s3_bucket",
        "namespace": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
        "location": [
          {
            "filepath": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
            "line": 14,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "tags",
              "owner"
            ],
            "location": {
              "filepath": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
              "line": 17,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "aws_s3_bucket.valid",
    "resource_namespace": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
    "resource_type": "aws_s3_bucket",
    "severity": "high",
    "resources": [
      {
        "id": "aws_s3_bucket.valid",
        "type": "aws_s3_bucket",
        "namespace": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
        "location": [
          {
            "filepath": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
            "line": 5,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "tags",
              "classification"
            ],
            "location": {
              "filepath": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
              "line": 8,
              "column": 3
            }
          },
          {
            "path": [
              "tags",
              "owner"
            ],
            "location": {
              "filepath": "spec/rules/REQUIRED_S3_BUCKET_TAGS/inputs/s3_bucket_tags.tf",
              "line": 8,
              "column": 3
            }
          }
        ]
      }
    ]
  }
]